### PR TITLE
Picking faceless mesh

### PR DESCRIPTION
**Pull request by CedricGuillemet**
Follow up on email discussion regarding 'picking a unique model (point cloud shape)'

In the mesh picking check that at least 1 submesh can be picked by ray-triangle otherwise falls back to bbox intersection.

### CODE REVIEW

File: src/Meshes/abstractMesh.ts [hunk 1]
```diff
@@ -1535,6 +1535,32 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
...
+                break;
+            }
+        }
+
+        // no sub mesh support intersection, fallback to BBox that has already be done
+        if (!anySubmeshSupportInterect) {
+            pickingInfo.hit = true;
+            pickingInfo.pickedMesh = this;
+            pickingInfo.distance = Vector3.Distance(ray.origin, boundingInfo.boundingSphere.center);
+            pickingInfo.subMeshId = 0;
```
Comment (Popov72) [inline src/Meshes/abstractMesh.ts:1562]: Maybe setting `subMeshId` to -1 would be better as there's really no submesh intersecting the raw?

File: src/Meshes/abstractMesh.ts [hunk 2]
```diff
@@ -1535,6 +1535,32 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
 
         var subMeshes = this._scene.getIntersectingSubMeshCandidates(this, ray);
         var len: number = subMeshes.length;
+
+        // Check if all submeshes are using a material that don't allow picking (point/lines rendering)
+        // if no submesh can be picked that way, then fallback to BBox picking
+        var anySubmeshSupportInterect = false;
```
Comment (Popov72) [inline src/Meshes/abstractMesh.ts:1541]: nit: typo `anySubmeshSupportIntersect`

**Comment by sebavan** [comment]
Love it, just wonder if there are any ways to prevent the double loop with caching or by doing some checks as part of the former one as I am afraid of the perf impact ???

File: src/Meshes/abstractMesh.ts (hunk 2 above)
Comment (CedricGuillemet) [reply src/Meshes/abstractMesh.ts]: Good catch!

**Comment by CedricGuillemet** [comment]
@sebavan unfortunately, I have to loop on submeshes :( Is `getMaterial()` from submesh that costly?

**Comment by deltakosh** [comment]
I think it is ok as the loop will break  immediatly most of the time

**Pull request merged**